projects/pgai/pgai/vectorizer/parsing.py

```diff
@@ -8,11 +10,17 @@
 
 from pgai.vectorizer.loading import LoadedDocument
 
+# Thread pool for CPU-intensive parsing operations
+max_workers = int(os.getenv("PARSING_MAX_WORKERS", 4))
+_PARSING_EXECUTOR = ThreadPoolExecutor(
```

Comment:
I mentioned [here](https://github.com/timescale/pgai/pull/802#discussion_r2154719913), but why not use a `ProccessPoolExecutor`?